lib/ansible/modules/cloud/docker/docker_image.py

```diff
@@ -467,12 +467,15 @@ def push_image(self, name, tag=None):
             self.results['changed'] = True
             if not self.check_mode:
                 status = None
+                changed = False
                 try:
                     for line in self.client.push(repository, tag=tag, stream=True, decode=True):
                         self.log(line, pretty_print=True)
                         if line.get('errorDetail'):
                             raise Exception(line['errorDetail']['message'])
                         status = line.get('status')
+                        if status == 'Pushing':
+                            changed = True
```

Comment:
Why not simply set `self.results['changed'] = False` here, instead of introducing `changed`?

Reply:
We are receiving many lines while streaming the progress bars for all layers. Assigning to a local variable should be cheaper due to a short lookup chain.
Setting `self.results['changed'] = False` at the top and `self.results['changed'] = True` here - in the loop - might be cleaner to read.

Reply:
Using a local variable in the loop makes sense. I would still try to localize its use a bit more, i.e. move the `changed = False` directly before the `for` loop (inside the `try` block), and the `self.results['changed'] = changed` directly after the loop (before the `except:`). That way, it's easier to see what happens with that variable.

Reply:
Done, thanks for the feedback.